auth/callback/interface.go

```diff
@@ -0,0 +1,28 @@
+package callback
+
+import (
+	"fmt"
+)
+
+type Unsupported struct {
+	Callback Interface
+}
+
+func (uc *Unsupported) Error() string {
+	return fmt.Sprintf("Unsupported callback <%T>: %v", uc.Callback, uc.Callback)
+}
+
+type Interface interface {
```

Comment:
Curious, what's the advantage of using marker interface *Interface*  type vs *interface{}* type ? Still need type assertion i see.